src/main/java/edu/kit/ipd/crowdcontrol/objectservice/quality/ratingQuality/RatingQualityByDistribution.java

```diff
@@ -0,0 +1,98 @@
+package edu.kit.ipd.crowdcontrol.objectservice.quality.ratingQuality;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.RatingRecord;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by lucaskrauss at 19.01.2016
+ */
+public class RatingQualityByDistribution implements RatingQualityStrategy {
+
+
+    /**
+     * @param ratings        list of RatingRecords whose quality will be rated
+     * @param maximumQuality
+     * @param minimumQuality
+     * @return
+     * @throws IllegalArgumentException
+     */
+    @Override
+    public Map<RatingRecord, Integer> identifyRatingQuality(List<RatingRecord> ratings, int maximumQuality, int minimumQuality) throws IllegalArgumentException {
+
+        Map<RatingRecord, Integer> map = new HashMap<>();
+
+        // Bucket sorts given ratings
+        LinkedList<RatingRecord>[] sortedRatings = new LinkedList[10];
+        for (RatingRecord rating : ratings) {
+
+            if (rating.getRating() < 0 || rating.getRating() > 9) {
+                throw new IllegalArgumentException("Error! Illegal rating value in " + this.getClass() + "! Expected value from 0 to 9" +
+                        ", but got :" + rating.getRating());
+            } else if (sortedRatings[rating.getRating()] == null) {
+                sortedRatings[rating.getRating()] = new LinkedList<RatingRecord>();
+            }
+            sortedRatings[rating.getRating()].add(rating);
+        }
```

Comment:
i mean bucket sort is cool and all, but `ratings.stream.collect(Collectors.groupBy(record -> record.getRating(), Collectors.toList());` would do the job and is bug-free